### PR TITLE
Remove unused argument of 'groupJobsByParent' in cronjob controller

### PR DESCRIPTION
**Pull request by junxu**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by junxu** [comment]
I signed it!

**Comment by junxu** [comment]
I signed it!

**Comment by junxu** [comment]
I signed it!

File: pkg/controller/cronjob/utils_test.go [hunk 1]
```diff
@@ -163,35 +163,19 @@ func TestGroupJobsByParent(t *testing.T) {
...
 		js := []batchv1.Job{}
-		jobsBySj := groupJobsByParent(sjs, js)
+		jobsBySj := groupJobsByParent(js)
 		if len(jobsBySj) != 0 {
 			t.Errorf("Wrong number of items in map")
 		}
 	}
 
 	{
-		// Case 2: there is one controller with no job.
```
Comment (janetkuo) [inline pkg/controller/cronjob/utils_test.go]: shall we keep the no jobs case?

**Comment by janetkuo** [comment]
@k8s-bot ok to test

File: pkg/controller/cronjob/utils_test.go (hunk 1 above)
Comment (junxu) [inline pkg/controller/cronjob/utils_test.go]: I think this is same as case 1, so i delete it.

**Review by janetkuo — APPROVED** [review]

**Pull request merged**